installer/functions

```diff
@@ -153,6 +201,35 @@ validate_name() {
     return 0
 }
 
+# Returns the mountpoint a path is on. The path doesn't need to exist.
+# $1: the path to check
+# outputs on stdout
+getmountpoint() {
+    mp="`readlink -m -- "$1"`"
+    while ! stat -c '%m' "$mp" 2>/dev/null; do
+        mp="${mp%/*}"
+    done
+}
```

Comment:
`getmountpoint /xyz` will result in an infinite loop (`/xyz` is stripped to an empty string).

Possible workaround: `stat -c '%m' "${mp:-/}"` (`getmountpoint ""` will fail as expected, due to `readlink`)